### PR TITLE
indexer: index more details about blocks and transactions

### LINKED ISSUE #1321: Return correct date for old blocks
Either by indexing the block headers or by other methods, we should provide a correct output to calls such as:

`curl https://api.vocdoni.net/v2/chain/blockToDate/500000`

### LINKED ISSUE #1349: bug: cannot get firsts blocks information
**Describe the bug**

On STG I cannot retrieve the information of firsts blocks, for example 

```bash
➜ curl 'https://api-stg.vocdoni.net/v2/chain/blocks/1' -s | jq 
{
  "error": "block not found",
  "code": 4008
}
```

But yes for latests block

```bash
➜ curl 'https://api-stg.vocdoni.net/v2/chain/blocks/1119027' -s | jq
```

`Dev` seems working

### LINKED ISSUE #1358: feature: return tx subtype on tx summary
**Describe the feature**

Return tx subtype on a transaction summary on:

`chain/transactions/page/0`
`chain/blocks/1374876/transactions/page/0`

**Motivation**

When getting a list of txs on the explorer, a nice [feature requested](https://github.com/vocdoni/explorer/issues/78) is to show the tx subtype instead of the tx type.

**Constraints & Assumptions**

Maybe can be also implemented on fees endpoints  because they have tx info and the tx type. 

`/chain/fees/page/:page`
`/chain/fees/reference/:reference/page/:page`
`/chain/fees/type/:type/page/:page`
`/accounts/:accountID/fees/page/:page`

### LINKED ISSUE #1370: API: transactions return the account
The information returned by the transactions endpoint to get the details of a transaction, the API backend should add a new field "account" which points to the address signing the transaction.

Currently, we do return the signature, so we can extract the address.

Example with: https://api.vocdoni.io/v2/chain/transactions/2414592/0

```json
{
  "tx": {
    "collectFaucet": {
      "txType": "TX_UNKNOWN",
      "faucetPackage": {
        "payload": "CIeH3RkSFA09aBdi+e2pVB+ras+xfFACW2MLGGQ=",
        "signature": "0j0vNTSzJo7NBxDIt++asacf7bW15s6KtzOzHMd8E1M0IIBMVHBUg0AMdMPZhTMevSN2J7wN6gcYDI5TfvsogAE="
      },
      "nonce": 16
    }
  },
  "txInfo": {
    "transactionNumber": 33680,
    "transactionHash": "76057f3845cf9e34e3cc35c5c08d598b61f11d46d4845706bb67f7b3d2612444",
    "blockHeight": 2414592,
    "transactionIndex": 0,
    "transactionType": "collectFaucet"
  },
  "signature": "82bdce114c0399a051d69b9ada79d0ab63d2540bc90025b3fe09cc6d73a3ce4752f00f7e46c01d089327818d4ebf1fb96c692897fe50a7ded9525048c9c021fc1b",
  "account": "0x123456789..."
}
```

This way, the explorer (@selankon), on each transaction, can show and link the account creating it.

Consider that some transactions (vote) won't have this information.

### PR DESCRIPTION
**Pull request by altergui**
needed since we started pruning the blockstore

### CODE REVIEW

**Comment by altergui** [comment]
@mvdan wdyt in general?

and in particular, TestRestoreBackupAndMigrate is failing, since i changed the db columns. what's the best fix? recreate the sqlite-backup-0009.sql.zst ?

i understand this would not happen if i avoided modifying 0006_create_table_blocks.sql and instead created a 0012_add_columns_to_table_blocks.sql , which i can also do.

File: api/helpers.go [hunk 1]
```diff
@@ -51,6 +52,38 @@ func (a *API) sendTx(tx []byte) (*cometcoretypes.ResultBroadcastTx, error) {
 	return resp, nil
 }
 
+func (a *API) getBlockFromAppOrIndexer(height int64) (*Block, error) {
+	if tmblock := a.vocapp.GetBlockByHeight(height); tmblock != nil {
+		return &Block{
+			Block: comettypes.Block{
+				Header:     tmblock.Header,
+				Data:       tmblock.Data,
+				Evidence:   tmblock.Evidence,
```
Comment (p4u) [inline api/helpers.go]: Do we need evidence?

Comment (altergui) [reply api/helpers.go]: i haven't touched this, it's just the previous code, and i'd leave it as-is.

but better yet, i've simplified this and now just use tmblock altogether

Comment (altergui) [reply api/helpers.go]: https://github.com/vocdoni/vocdoni-node/compare/dbdd78ff508f48e60a450f43d5a5d139a50ef990..711f82e200237bdd406173d8d5874eb462b3a96a

**Comment by mvdan** [comment]
Yes, please add a new migration step. As far as I can tell, and from what we agreed with @p4u, we are treating the database as stable and we should avoid any breaking changes that force us to delete it and re-index everything. In most cases, like adding new columns, it should be easy to do it as a new step anyway.

File: api/chain.go [hunk 2]
```diff
@@ -422,9 +421,16 @@ func (a *API) chainEstimateDateHandler(_ *apirest.APIdata, ctx *httprouter.HTTPC
 		return err
 	}
 	timestamp := a.vocinfo.HeightTime(height)
+	if timestamp == nil {
```
Comment (mvdan) [inline api/chain.go]: FWIW I would keep HeightTime to returning a non-pointer, and you can use the IsZero method to check if it's valid/present or not.

File: api/helpers.go [hunk 3]
```diff
@@ -51,6 +52,33 @@ func (a *API) sendTx(tx []byte) (*cometcoretypes.ResultBroadcastTx, error) {
 	return resp, nil
 }
 
+func (a *API) getBlockFromAppOrIndexer(height int64) (*Block, error) {
```
Comment (mvdan) [inline api/helpers.go]: why is it useful to query either the app or the indexer? wouldn't it be enough to always just query one? if this is necessary at least I would document it

Comment (altergui) [reply api/helpers.go]: good point, maybe we could refactor this, but for now i just added a comment

**Comment by altergui** [comment]
TestRestoreBackupAndMigrate is still failing, with the new migration 0013, complaining about a missing column. digging in :detective:

File: api/helpers.go (hunk 3 above)
Comment (altergui) [reply api/helpers.go]: actually, in the end, i split up the crucial hotfix (#1339) and removed this "getBlockFromAppOrIndexer" completely

**Comment by altergui** [comment]
> TestRestoreBackupAndMigrate is still failing, with the new migration 0013, complaining about a missing column. digging in 🕵️

this was failing due to my buggy migration 0013. fixed, all tests are passing now

**Review by p4u — APPROVED** [review]

**Pull request merged**